### PR TITLE
refactor: p2p complexity

### PR DESCRIPTION
**Pull request by faustbrian**
## Proposed changes

The complexity of the P2P Interface was quite high **(and still is)** so this PR tries to reduce this a bit.

## Types of changes

- [x] Refactoring (improve a current implementation without adding a new feature or fixing a bug)

## Checklist

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/developers/guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/core-p2p/lib/monitor.js [hunk 1]
```diff
@@ -15,54 +15,62 @@ const Peer = require('./peer')
...
     try {
       // TODO: for tests that involve peers we need to sync them
       if (process.env.ARK_ENV !== 'test') {
         await this.discoverPeers()
         await this.cleanPeers()
       }
 
-      if (Object.keys(this.peers).length < this.config.peers.list.length - 1 && process.env.ARK_ENV !== 'test') {
-        this.config.peers.list
+      if (Object.keys(this.peers).length < config.peers.list.length - 1 && process.env.ARK_ENV !== 'test') {
```
Comment (alexbarnsley) [inline packages/core-p2p/lib/monitor.js]: Shouldn't this be `this.config`? I don't see config passed into updateNetworkStatus, or declared outside of the class.

File: packages/core-p2p/lib/manager.js [hunk 2]
```diff
@@ -1,198 +0,0 @@
...
-  async updateNetworkStatus () {
-    await this.monitor.updateNetworkStatus()
-  }
-
-  /**
-   * Download blocks from a random peer.
-   * @param  {Number}   fromBlockHeight
-   * @return {Object[]}
-   */
-  async downloadBlocks (fromBlockHeight) {
```
Comment (alexbarnsley) [inline packages/core-p2p/lib/manager.js]: This needs moving to `monitor.js` as it catches an exception in getRandomPeers, or at least it used to. Related to this Exception spam: https://github.com/ArkEcosystem/core/issues/852

**Comment by alexbarnsley** [comment]
I meant to comment - not tested yet, as I think there may be a couple of issues. Check the config => `this.config` comment i made, then I will test whether the downloadBlocks try/catch removal is still an issue.

File: packages/core-p2p/lib/monitor.js (hunk 1 above)
Comment (faustbrian) [reply packages/core-p2p/lib/monitor.js]: `config` is the `container.resolvePlugin('config')`

File: packages/core-p2p/lib/manager.js (hunk 2 above)
Comment (faustbrian) [reply packages/core-p2p/lib/manager.js]: Will check that later, should have been fine as it handles the exception itself already 🤔

Comment (faustbrian) [reply packages/core-p2p/lib/manager.js]: The `getRandomDownloadBlocksPeer` was throwing the exception so I moved it inside the `try catch` block in `downloadBlocks`.

File: packages/core-p2p/lib/monitor.js (hunk 1 above)
Comment (alexbarnsley) [reply packages/core-p2p/lib/monitor.js]: I guess i needed sleep, I looked for it at the top and didn't see it 🤦‍♂️

File: packages/core-p2p/lib/manager.js (hunk 2 above)
Comment (alexbarnsley) [reply packages/core-p2p/lib/manager.js]: Will test now, thanks!

**Pull request merged**